app/models/invoice.rb

```diff
@@ -22,19 +22,24 @@ class Invoice < ApplicationRecord
 
   has_one_attached :file
 
-  monetize :amount_cents
-  monetize :vat_amount_cents
-  monetize :credit_amount_cents
-  monetize :total_amount_cents
+  # NOTE: currency will be added by rails, but to enable monetize, we need to expose it before
+  def currency
+    super
+  end
+
+  monetize :amount_cents, with_model_currency: :currency
+  monetize :vat_amount_cents, with_model_currency: :currency
+  monetize :credit_amount_cents, with_model_currency: :currency
+  monetize :total_amount_cents, with_model_currency: :currency
 
   # NOTE: Readonly fields
-  monetize :sub_total_vat_excluded_amount_cents, disable_validation: true, allow_nil: true
-  monetize :sub_total_vat_included_amount_cents, disable_validation: true, allow_nil: true
-  monetize :coupon_total_amount_cents, disable_validation: true, allow_nil: true
-  monetize :credit_note_total_amount_cents, disable_validation: true, allow_nil: true
-  monetize :charge_amount_cents, disable_validation: true, allow_nil: true
-  monetize :subscription_amount_cents, disable_validation: true, allow_nil: true
-  monetize :wallet_transaction_amount_cents, disable_validation: true, allow_nil: true
+  monetize :sub_total_vat_excluded_amount_cents, disable_validation: true, allow_nil: true, with_model_currency: :currency
```

Comment:
Layout/LineLength: Line is too long. [122/120] (https://rubystyle.guide#max-line-length)